scripts/WAM2-Layers Source Region Designer.py

```diff
@@ -0,0 +1,288 @@
+            #GENERAL SETTINGS
+#=======================================
+
+#Output directory - Set here where the output is stored
+output_dir = '...'
+
+#Specify latitude and longitude - same as input data to WAM2Layers
+lat_max = 80
+lat_min = -80
+
+lon_max = 180
+lon_min = -180
+
+step = 0.25 #0.25 degree intervals (ERA5 data)
+
+#Ranges
+latitude = np.arange(lat_min,lat_max+step,step) #check ranges based on your input
+longitude = np.arange(lon_min,lon_max,step)
+
+#=======================================
+            #METHOD SELECTOR
+#=======================================
+selector = 1    #1 = Method 1, 2 = Method 2, 3 = Method 3. 
+
+#======================================================================================================================
+
+#=======================================
+        #METHOD SPECIFIC SETTINGS
+#=======================================
+
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+#           M E T H O D  1
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+#Create an overview of available regions from: https://regionmask.readthedocs.io/en/stable/defined_scientific.html
+available_regions = regionmask.defined_regions.giorgi #<-- specify available regions source (giorgi or other)
+
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+#           M E T H O D  2
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+shapefile = '...'    #specify a shapefile
+
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+#           M E T H O D  3
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+# Define the center point and distance in degrees
+
+#Center point
+center_lat = -2         
+center_lon = -58.75
+    
+distance = 20               #Distance of the square around the center point, in degrees
+
+
+'''
+
+PLEASE NOTE: region ID's (what specific region or regions) to export as a source region, need to be specified within the code itself. 
+A visual representation of the region is generated to show intermediate steps / the final product. 
+
+'''
+
+
+#%%
+
+'''
+
+METHOD 1 - Using the available regions from the 'region mask' python package
+ 
+'''
+
+if selector == 1:
+        
+    #Visualisation of the regions and their ID's --> select region
+    text_kws = dict(
+        bbox=dict(color="none"),
+        path_effects=[pe.withStroke(linewidth=2, foreground="w")],
+        color="#67000d",
+        fontsize=8,
+    )
+    
+    ax = available_regions.plot(
+        projection=ccrs.Robinson(), add_ocean=True, text_kws=text_kws
+    )
+    
+    ax.set_global()   
+    plt.show()
+    
+    
+    #create (a) mask(s) of the required region(s):
+    mask = regionmask.defined_regions.giorgi.mask_3D(longitude, latitude)   #<-- specify available regions source
+    
+    #Specify region ID - what to store
+    region_id = 2                                                           #single mask id
+    region_ids = [1]                                                        #multiple regions in one mask [id-1,id-1,...]
```

Comment:
are these also settings? If so, please move to settings

Reply:
The thing with this is that it is dependent on the user. Whilst developing I noticed that it is useful to have a feedback with plotting and looking on whether you have selected the correct things. So therefore, I had intermediate plotting. Maybe, indeed as @Peter9192 suggested at the start, I could implement this more as a jupyter notebook script. In that way, you can have intermediate plotting to really see whether the thing you plot is what you want... Or, we assume that the user knows which region_id or ids to select...

Reply:
Sounds like you could also have a separate function to plot/preview the region you're cropping